recipes/jemalloc/all/conanfile.py

```diff
@@ -66,25 +77,24 @@ def configure(self):
         if self.settings.compiler == "Visual Studio" and self.settings.compiler.version != "15":
             # https://github.com/jemalloc/jemalloc/issues/1703
             raise ConanInvalidConfiguration("Only Visual Studio 15 2017 is supported.  Please fix this if other versions are supported")
-        if self.options.shared:
-            del self.options.fPIC
-        if not self.options.enable_cxx:
-            del self.settings.compiler.libcxx
-            del self.settings.compiler.cppstd
         if self.settings.build_type not in ("Release", "Debug", None):
             raise ConanInvalidConfiguration("Only Release and Debug build_types are supported")
         if self.settings.compiler == "Visual Studio" and self.settings.arch not in ("x86_64", "x86"):
             raise ConanInvalidConfiguration("Unsupported arch")
         if self.settings.compiler == "clang" and tools.Version(self.settings.compiler.version) <= "3.9":
             raise ConanInvalidConfiguration("Unsupported compiler version")
 
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        os.rename("{}-{}".format(self.name, self.version), self._source_subfolder)
+    def package_id(self):
+        if not self.settings.build_type:
+            self.info.settings.build_type = "Release"
```

Comment:
@madebr I kept this part, but I don't understand it.

Reply:
I don't remember why it is there.
Looks like it was there from my first commit.
You can remove it.